### PR TITLE
Verify Mac artifact codesigning on x64 and arm64

### PR DESCRIPTION
**Pull request by jmagman**
Intel x64 and ARM Mac artifacts are uploaded separately.  Run codesign verification on Macs of both architectures.

Would prevent arm packages from being shipped without codesigning, as happened in 3.3.4 and 3.7: https://github.com/flutter/flutter/issues/111764#issuecomment-1275539149

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -3215,7 +3215,24 @@ targets:
...
+  - name: Mac_x64 verify_binaries_codesigned
+    bringup: true
+    enabled_branches:
+      - flutter-\d+\.\d+-candidate\.\d+
+    recipe: flutter/flutter
+    timeout: 60
+    properties:
+      dependencies: >-
+        [
+          {"dependency": "xcode", "version": "14a5294e"}
```
Comment (christopherfujino) [inline .ci.yaml:3227]: this cipd package already exists for arm?

**Comment by XilaiZhang** [comment]
LGTM. umm shamefully speaking verify binary codesigned is a tech debt that historically and notoriously does not run on release branch due to git issue https://github.com/flutter/flutter/issues/116611

File: .ci.yaml (hunk 1 above)
Comment (jmagman) [reply .ci.yaml:3227]: Yes, it's fat and will work on either.

Comment (christopherfujino) [reply .ci.yaml:3227]: I shudder to think how big a fat xcode package is.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by jmagman** [comment]
> LGTM. umm shamefully speaking verify binary codesigned is a tech debt that historically and notoriously does not run on release branch(#116611 ) due to git issue (e.g. https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8790347300087503201/+/u/Verify_binaries_codesigned/verify_binaries_codesigned/test_stderr) I remember tracing it with Chrsitopher but it wasn't trivial to fix

@XilaiZhang how do you suggest I force these to run then?

**Comment by XilaiZhang** [comment]
> > LGTM. umm shamefully speaking verify binary codesigned is a tech debt that historically and notoriously does not run on release branch(#116611 ) due to git issue (e.g. https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8790347300087503201/+/u/Verify_binaries_codesigned/verify_binaries_codesigned/test_stderr) I remember tracing it with Chrsitopher but it wasn't trivial to fix
> 
> @XilaiZhang how do you suggest I force these to run then to test them before the next release?

Umm i am not sure how to surpass the git error in the luci build. Maybe Christopher would know more?

**Pull request merged**